api-definitions/common/file-pointer.yaml

```diff
@@ -0,0 +1,35 @@
+################################################################################
+# This schema defines the common ways to reference files and directories held in
+# the various supported storage media.
+################################################################################
+
+schemas:
+  LocalPath:
+    required: ['path']
+    properties:
+      path:
+        type: string
+        description: The name of the file on disk
+
+  S3Path:
```

Comment:
Why does it have to be S3? How about we make this API standard agnostic? `ObjectStoragePath` looks sufficient (or something along those lines) and will comply with S3 as well as AzureBlob for instance. We might need to introduce a `provider` property that specifies the type of object storage.

Or perhaps you can point me to where it was decided to use S3 only? Thanks!

Reply:
Interesting. This is probably one of those places where I have a mental word for a thing that I've never questioned. I was under the impression that `S3` had been adopted as the abstract API of which `Amazon S3` was the canonical, but not only, implementation. For example [IBM COS](https://cloud.ibm.com/apidocs/cos/cos-compatibility) documents itself as `S3 compatible`. I'm absolutely open to using a more generic name for this if `S3` is not the right generic label.

Reply:
I don't think we should be tied to S3, especially since other cloud providers like GCP and Azure are not compatible with it. Also S3 is a protocol not strictly an interface (I don't want to get into semantics; I'll lose that battle lol) but I hope you get my point. It's an object interface, like we have file and block. I maintain `ObjectStoragePath` which makes us agnostic or `ObjectPath`. Hope that makes sense.